File: tools/handshake_youtube.py

import os
from googleapiclient.discovery import build
from dotenv import load_dotenv

# Load environment variables
load_dotenv()

api_key = os.environ.get("YOUTUBE_API_KEY")

if not api_key:
    print("Error: Missing YOUTUBE_API_KEY in .env")
    exit(1)

try:
    youtube = build('youtube', 'v3', developerKey=api_key)
    # Simple query to check connection - search for "Google" channel
    request = youtube.channels().list(
        part="snippet",
        forUsername="Google"
    )
    response = request.execute()
    
    print("Connection Successful! YouTube API Responded.")
    print(f"Response Items: {len(response.get('items', []))}")

except Exception as e:
    print(f"YouTube Handshake Failed. Result: {e}")

File: tools/handshake_youtube.js

import dotenv from 'dotenv';
import path from 'path';

dotenv.config({ path: path.resolve(process.cwd(), '.env') });

const apiKey = process.env.YOUTUBE_API_KEY;

if (!apiKey) {
    console.error("Error: Missing YOUTUBE_API_KEY in .env");
    process.exit(1);
}

async function testConnection() {
    try {
        const url = `https://www.googleapis.com/youtube/v3/channels?part=snippet&forUsername=Google&key=${apiKey}`;
        const response = await fetch(url);

        if (!response.ok) {
            const err = await response.text();
            throw new Error(`API Error: ${response.status} ${err}`);
        }

        const data = await response.json();
        console.log("Connection Successful! YouTube API Responded.");
        console.log(`Items found: ${data.items ? data.items.length : 0}`);

    } catch (e) {
        console.error(`YouTube Handshake Failed: ${e.message}`);
        process.exit(1);
    }
}

testConnection();
